packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -532,6 +541,7 @@ export const DatePickerMixin = (subclass) =>
         !this._inputValue ||
         (!!this._selectedDate && this._inputValue === this._getFormattedDate(this.i18n.formatDate, this._selectedDate));
       const minMaxValid = !this._selectedDate || dateAllowed(this._selectedDate, this._minDate, this._maxDate);
+      const isDateAvailableValid = this.isDateAvailable ? this.isDateAvailable(this._selectedDate) : true;
```

Comment:
Since `dateAllowed` is capable of checking for whether the date should be disabled, it seems unnecessary to call `isDateAvailable` separately. Instead, I would suggest to pass the function into `dateAllowed`, and generalize the constant name from `minMaxValid` to something more general like `isDateValid`.

Reply:
Much cleaner.